src/c++/perf_analyzer/data_loader.h

```diff
@@ -39,10 +39,6 @@ class DataLoader {
  public:
   DataLoader(size_t batch_size);
 
-  /// Returns the total number of data steps that can be supported by a
-  /// non-sequence model.
-  size_t GetTotalStepsNonSequence() { return max_non_sequence_step_id_; }
```

Comment:
Can you high-level explain why this API is no longer needed?

Reply:
It is related to [this](https://github.com/triton-inference-server/client/pull/259#discussion_r1132841878) comment, although it applies to both shm and non-shm cases.

The API was basically returning the number of steps we could take without going past the end of the custom data. For the example of [1,2,3,4,5] with a batch size of 2, that function would have returned 2. 

It was used [here](https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/infer_context.cc#L178). That is how we ended up with [1,2] and [3,4] in my example above instead of properly walking and wrapping around.